fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -220,10 +225,14 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 
 	struct Reader : IThreadPoolReceiver {
 		DB& db;
+		ThreadStatisticsState threadStatState;
 
-		explicit Reader(DB& db) : db(db) {}
+		explicit Reader(DB& db, std::string name) : db(db), threadStatState(name) {}
 
-		void init() override {}
+		void init() override {
+			threadMonitor(threadStatState);
+			uncancellable(recurring([this]() { threadMonitor(threadStatState); }, SERVER_KNOBS->ROCKSDB_THREADS_MONITOR_FREQUENCY, TaskPriority::FlushTrace));
```

Comment:
AFAIU, creating actors on background threads is unsupported. You could run TSAN to confirm, but I am pretty sure this is a data race.

Reply:
TSAN ran successfully